app/src/main/java/com/github/warnastrophy/core/model/GpsService.kt

```diff
@@ -0,0 +1,229 @@
+   */
+  @SuppressLint("MissingPermission")
+  override fun startLocationUpdates(locationClient: FusedLocationProviderClient) {
+    Log.e(TAG, "launching startLocationUpdates")
+    setLoading(true)
+
+    try {
+      val request =
+          LocationRequest.Builder(Priority.PRIORITY_HIGH_ACCURACY, AppConfig.positionUpdateDelayMs)
+              .build()
+
+      locationClient.requestLocationUpdates(
+          request,
+          object : LocationCallback() {
+            override fun onLocationResult(result: LocationResult) {
+
+              val pos = result.lastLocation
+              Log.e(TAG, "requestPosupd : $pos")
+              if (pos == null) {
+                setError("No location fix available")
+                setLoading(false)
+              } else {
+                updatePosition(LatLng(pos.latitude, pos.longitude))
+                setSuccess("Position updated")
+                setLoading(false)
+              }
+            }
+
+            override fun onLocationAvailability(availability: LocationAvailability) {
+              if (!availability.isLocationAvailable) {
+                setError("Location temporarily unavailable")
+                setLoading(false)
+              } else {
+                setSuccess("Location available")
+                setLoading(false)
+              }
+            }
+          },
+          Looper.getMainLooper())
+    } catch (e: SecurityException) {
+      setError("Location permission not granted!")
+      setLoading(false)
+    } catch (e: Exception) {
+      setError("Location update failed: ${e.message}")
+      setLoading(false)
+    }
+  }
+
+  /**
+   * Updates the position in the state with success.
+   *
+   * @param position The new GPS position.
+   */
+  private fun updatePosition(position: LatLng) {
+    _positionState.update { currentState ->
+      currentState.copy(
+          position = position, result = GpsResult.Success("Position updated"), errorMessage = null)
+    }
+  }
+
+  /**
+   * Sets an error message in the state.
+   *
+   * @param message The error message to set.
+   */
+  private fun setError(message: String) {
+    _positionState.update { currentState ->
+      currentState.copy(result = GpsResult.Failed, errorMessage = message)
+    }
+  }
+
+  /**
+   * Sets a success message in the state.
+   *
+   * @param message The success message to set.
+   */
+  private fun setSuccess(message: String) {
+    _positionState.update { currentState ->
+      currentState.copy(result = GpsResult.Success(message), errorMessage = null)
+    }
+  }
+
+  /**
+   * Updates the loading state.
+   *
+   * @param isLoading Whether the service is currently loading.
+   */
+  private fun setLoading(isLoading: Boolean) {
+    _positionState.update { currentState -> currentState.copy(isLoading = isLoading) }
+  }
+
+  /** Releases resources used by this service. Call when the service is no longer needed. */
+  fun close() {
+    serviceScope.cancel()
+  }
```

Comment:
We call `stopLocationUpdates()` here also.
```suggestion
  fun close() {
    stopLocationUpdates()
    serviceScope.cancel()  
  }
```